### PR TITLE
feat(rust): add reorg, disconnect, and undo parity (Q-RUST-PARITY-REORG-01)

### PR DESCRIPTION
**Pull request by 2tbmz9y2xt-lang**
## Summary

- Add full block disconnection, undo record management, and chain reorganization to the Rust node, achieving parity with the Go reference implementation
- New modules: `undo.rs`, `sync_disconnect.rs`, `sync_reorg.rs` — covering UTXO restoration, canonical chain rollback, fork-choice evaluation with dry-run validation, and mempool tx requeue/eviction
- Modified `blockstore.rs` (undo storage, chain work, side-chain block storage), `sync.rs` (pub(crate) fields, undo integration, rollback state), `txpool.rs` (evict_txids), `lib.rs` (module registration)

## Test plan

- [x] 122 existing tests pass (including new undo roundtrip, serialization edge cases, disconnect error paths, non-coinbase tx extraction)
- [x] Clippy clean (`-D warnings`)
- [x] `cargo fmt` applied
- [x] CI checks (Codacy coverage, clippy, test suite)

**Task:** Q-RUST-PARITY-REORG-01

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: clients/rust/crates/rubin-node/src/sync.rs [hunk 1]
```diff
@@ -221,6 +246,29 @@ impl SyncEngine {
...
+            best_known_height: self.best_known_height,
+        }
+    }
+
+    pub(crate) fn rollback_apply_block(&mut self, rb: SyncRollbackState) {
+        self.chain_state = rb.chain_state;
+        self.tip_timestamp = rb.tip_timestamp;
+        self.best_known_height = rb.best_known_height;
+        if let Some(bs) = self.block_store.as_mut() {
+            let _ = bs.truncate_canonical(rb.canonical_len);
```
[the remark this answers is not in the document]
Comment (2tbmz9y2xt-lang) [reply clients/rust/crates/rubin-node/src/sync.rs]: Fixed in `73ade69`: `rollback_apply_block` now uses `restore_canonical(rb.canonical_snapshot)` instead of `truncate_canonical(rb.canonical_len)`. `SyncRollbackState` captures a full `Vec<String>` clone of the canonical index at snapshot time. New `BlockStore::restore_canonical()` replaces the entire canonical array, so rollback works correctly even after disconnect has removed entries.
